pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -588,6 +588,17 @@ class ModelValidatorImpl implements ModelValidator {
                         }
                     }
                     map.variables.each { k, v ->
+                        // TODO: JENKINS-41746 - get rid of this hardcoding and use extensible validation instead once we can
+                        if (typeName == "dockerfile") {
+                            if ((k.key == "dir" || k.key == "filename") && v.toGroovy().contains(" ")) {
+                                errorCollector.error(k, Messages.ModelValidatorImpl_SpaceInDockerfilePath(k.key))
+                                valid = false
+                            }
+                            if (k.key == "filename" && (v.toGroovy().contains("/") || v.toGroovy().contains("\\"))) {
+                                errorCollector.error(k, Messages.ModelValidatorImpl_DirSeparatorInDockerfileName())
```

Comment:
wasn't this possible to do in 1.0? and are we breaking behavior by not allowing this?